custom_components/goodwe/number.py

```diff
@@ -47,6 +47,30 @@ async def _set_eco_mode_power(entity: InverterNumberEntity, value: int) -> None:
 
 
 NUMBERS = (
+    GoodweNumberEntityDescription(
+        key="battery_charge_limit",
+        name="Battery charge limit",
+        icon="mdi:battery-charging",
+        entity_category=EntityCategory.CONFIG,
+        unit_of_measurement=ELECTRIC_CURRENT_AMPERE,
+        getter=lambda inv: inv.read_setting('battery_charge_current'),
+        setter=lambda inv, val: inv.inverter.write_setting('battery_charge_current', val),
+        step=1,
+        min_value=0,
+        max_value=50,
+    ),
+    GoodweNumberEntityDescription(
+        key="battery_discharge_limit",
+        name="Battery discharge limit",
+        icon="mdi:battery-charging",
+        entity_category=EntityCategory.CONFIG,
+        unit_of_measurement=ELECTRIC_CURRENT_AMPERE,
+        getter=lambda inv: inv.read_setting('battery_discharge_current'),
+        setter=lambda inv, val: inv.inverter.write_setting('battery_discharge_current', val),
+        step=1,
+        min_value=0,
+        max_value=50,
```

Comment:
Actually, this limit is a bit problematic.
It may be +/- ok for high-voltage inverters like ET/EH, but AFAIK the older low-voltage inverter like ES/EM use way higher charge/discharge currents.
And I'm not sure how to properly differentiate there ...